internal/gcs-sidecar/bridge.go

```diff
@@ -448,6 +449,23 @@ func (b *Bridge) ListenAndServeShimRequests() error {
 					_ = sendWithContextCancel(ctx, sidecarErrChan, recverr)
 					return
 				}
+				// If this is a ContainerExecuteProcessResponse, notify
```

Comment:
Instead of this approach, we can use channels for capturing the process response.
Approach would be-
- Create a channel on which you write the response to when the same is received in `ListenAndServeShimRequests` method. 
- Same channel is used in `executeProcess` method. We listens on the channel with a deadline of 5 seconds. If you don't hear back then return an error message. If we hear back about another exec then we skip and keep waiting for next response.

With this approach, you won't need- `b.pending`, `b.pendingMu` and other polling logic.